### ISSUE TITLE
NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()'

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/focus-android#1789]
```
java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a n
                            ull object reference
                         E      at org.mozilla.focus.search.SearchEngineManager.getDefaultSearchEngine(SearchEngineManager.java:288)
                         E      at org.mozilla.focus.search.SearchEngineListPreference.refreshSearchEngineViews(SearchEngineListPreference.java:68)
                         E      at org.mozilla.focus.search.SearchEngineListPreference.refetchSearchEngines(SearchEngineListPreference.java:57)
                         E      at org.mozilla.focus.settings.SettingsFragment.refetchSearchEngines(SettingsFragment.java:216)
                         E      at org.mozilla.focus.settings.SettingsFragment.onResume(SettingsFragment.java:204)
                         E      at android.app.Fragment.performResume(Fragment.java:2554)
                         E      at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1322)
                         E      at android.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1557)
                         E      at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1618)
                         E      at android.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2196)
                         E      at android.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2142)
                         E      at android.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:868)
                         E      at android.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:812)
                         E      at android.app.Activity.onBackPressed(Activity.java:3009)
                         E      at android.support.v4.app.FragmentActivity.onBackPressed(FragmentActivity.java:175)
                         E      at org.mozilla.focus.activity.SettingsActivity.onOptionsItemSelected(SettingsActivity.java:51)
                         E      at android.app.Activity.onMenuItemSelected(Activity.java:3451)
                         E      at android.support.v4.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:368)
                         E      at android.support.v7.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:195)
                         E      at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:108)
                         E      at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:108)
                         E      at android.support.v7.widget.ToolbarWidgetWrapper$1.onClick(ToolbarWidgetWrapper.java:187)
                         E      at android.view.View.performClick(View.java:6294)
                         E      at android.view.View$PerformClick.run(View.java:24770)
                         E      at android.os.Handler.handleCallback(Handler.java:790)
                         E      at android.os.Handler.dispatchMessage(Handler.java:99)
                         E      at android.os.Looper.loop(Looper.java:164)
                         E      at android.app.ActivityThread.main(ActivityThread.java:6494)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by pocmo** [comment]
I saw this crash multiple times locally. I have no steps to reproduce but it happens frequently after changing default search engines and going to the remove search engine UI and back.

**Comment by pocmo** [comment]
This should get fixed before 4.0 release.

**Comment by mcomella** [comment]
Dupe #1819?
